File: _CodeTopics/LeetCode/201-400/000389/000389.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """

        def str_to_dict(s):
            dic = {}
            for ch in s:
                if ch in dic:
                    dic[ch] += 1
                else:
                    dic[ch] = 1
            return dic
        
        dicS, dicT = str_to_dict(s), str_to_dict(t)
        for k, v in dicT.items():
            if k not in dicS:
                return k
            elif v > dicS[k]:
                return k
        
"""
https://leetcode-cn.com/submissions/detail/131939320/

54 / 54 个通过测试用例
状态：通过
执行用时: 20 ms
内存消耗: 13.1 MB

执行用时：20 ms, 在所有 Python 提交中击败了81.42%的用户
内存消耗：13.1 MB, 在所有 Python 提交中击败了16.14%的用户
"""

File: _CodeTopics/LeetCode/201-400/000389/tran/000389.cpp
class Solution {
public:
    unordered_map<char, int> str_to_dic(string s) {
        unordered_map<char, int> dic;
        for (auto ch : s) {
            if (dic.find(ch) != dic.end()) {
                dic[ch]++;
            }
            else {
                dic[ch] = 1;
            }
        }
        return dic;
    }

    char findTheDifference(string s, string t) {
        unordered_map<char, int> dicS, dicT;
        dicS = str_to_dic(s);
        dicT = str_to_dic(t);
        for (auto kv = dicT.begin(); kv != dicT.end(); kv++) {
            if (dicS.find(kv->first) == dicS.end()) {
                return kv->first;
            }
            else if (kv->second > dicS[kv->first]) {
                return kv->first;
            }
        }
        return -1;
    }
};

/*
https://leetcode-cn.com/submissions/detail/131964426/

54 / 54 个通过测试用例
状态：通过
执行用时: 16 ms
内存消耗: 7.4 MB

执行用时：16 ms, 在所有 C++ 提交中击败了12.97%的用户
内存消耗：7.4 MB, 在所有 C++ 提交中击败了5.03%的用户
*/
